src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTest.cs

```diff
@@ -86,12 +88,20 @@ internal void Execute(DotNetSdkHelper dotNetHelper, string testRoot, string[]? f
         if (Commands.HasFlag(DotNetSdkActions.PublishComplex))
         {
             dotNetHelper.ExecutePublish(projectDirectory, selfContained: false);        
-            dotNetHelper.ExecutePublish(projectDirectory, selfContained: true, TargetRid);
-            dotNetHelper.ExecutePublish(projectDirectory, selfContained: true, $"linux-{TargetArchitecture}");
+            dotNetHelper.ExecutePublish(projectDirectory, TargetRid, selfContained: true);
+            dotNetHelper.ExecutePublish(projectDirectory, $"linux-{TargetArchitecture}", selfContained: true);
         }
         if (Commands.HasFlag(DotNetSdkActions.PublishR2R))
         {
-            dotNetHelper.ExecutePublish(projectDirectory, selfContained: true, $"linux-{TargetArchitecture}", trimmed: true, readyToRun: true);
+            dotNetHelper.ExecutePublish(projectDirectory, $"linux-{TargetArchitecture}", selfContained: true, trimmed: true, readyToRun: true);
+        }
+        if (Commands.HasFlag(DotNetSdkActions.PublishAot))
+        {
+            // Non-portable bundled AOT Compiler.
+            if (isNonPortable && !dotNetHelper.IsMonoRuntime)
```

Comment:
Ideally, this would be skipped at the test level with `SkippableFact`. This should be possible because, at a minimum, the SDK root location is known via environment variable. If that approach doesn't work out, it'd be good to at least write something to the output, via `DotNetHelper.OutputHelper`, that this is being skipped and the reason why.

Reply:
> Ideally, this would be skipped at the test level with SkippableFact. 

The project has no reference to a library that provides `SkippableFact`. What implementation should be used? Should it be a project reference, or do we want to include the sources into the project?

> (isNonPortable && !dotNetHelper.IsMonoRuntime)

Currently, this check is made when the test class was instantiated.

To work nicely with `SkippableFact` we probably want to have a `static` check and for that `ScenarioTestFixture` would be refactored into a `static class`.

Reply:
Actually, it looks like there's already an existing pattern for skipping tests in this repo. It uses the `Trait` attribute. For example, it skips tests based on platform with this: https://github.com/dotnet/scenario-tests/blob/f1715bf842371d15ecb5fc52d7dc87ab2e47df30/src/Microsoft.DotNet.ScenarioTests.Common/Program.cs#L235 ([example usage of trait](https://github.com/dotnet/scenario-tests/blob/f1715bf842371d15ecb5fc52d7dc87ab2e47df30/src/Microsoft.DotNet.ScenarioTests.SdkTemplateTests/SdkTemplateTests.cs#L114-L120))

It'd be good if that pattern could be used here.

Reply:
ok, I'll follow that pattern. The trait would be something like: `BundledAotCompiler`?

Reply:
Yeah, something like that.

Reply:
I hadn't used `XunitFilters` before. From looking at it closer, the `IncludedTraits` requires all tests to have that trait, so I instead of an included trait (`"BundledAotCompiler"`), I've implemented it using excluded traits (`!"Portable", !"Mono"`).

@mthalman ptal.